compiler/rustc_borrowck/src/lib.rs

```diff
@@ -1465,6 +1470,9 @@ impl<'cx, 'tcx> MirBorrowckCtxt<'cx, 'tcx> {
                 );
             }
             Operand::Move(place) => {
+                // Check if moving from this place makes sense.
+                self.check_movable_place(location, place);
```

Comment:
Why is `check_movable_place` called from two places?
Can it be centralized to one place?

Reply:
We want to report move errors from promoted, but the different borrowck analyses are still run manually for them. This is the least-invasive way to keep the behaviour.
The "proper" way to unify both cases would be https://github.com/rust-lang/rust/pull/111693 or something like it.